common/utils/file.go

```diff
@@ -18,7 +18,7 @@ func ReadSecretFile(filePath string) (pulumi.StringPtrOutput, error) {
 }
 
 func WriteStringCommand(content pulumi.StringInput, filePath string) pulumi.StringInput {
-	return pulumi.Sprintf(`cat <<EOF > %s
+	return pulumi.Sprintf(`sh -c 'cat <<EOF > %s
```

Comment:
Why do we need this `sh` wrapper? The commands are already run through a shell.

Reply:
When running `sudo` with `cat <<EOF > %s`, sudo applies only to `cat` and not to the redirection. This is for example required to update `datadog.yaml`.

Reply:
I think it would be cleaner to have a `sudo` parameter and use `tee` to write to a file:
```
	return pulumi.Sprintf(`cat - | %s tee %s > /dev/null`, sudo, filePath)
```

Reply:
Code updated.